### ISSUE TITLE
null value fields in payload should be skipped

### ISSUE THREAD

**Issue by karthiknadig** [openlawlibrary/pygls#145]
I am not sure if the LSP spec specifies if you can use `null` for optional fields in response or notification. But this seems to cause issues with VS Code.

Here is an example, consider this response content from `pygls`:
```json
[
    {
        "name": "SampleClass",
        "kind": 5,
        "range": {
            "start": {
                "line": 17,
                "character": 0
            },
            "end": {
                "line": 19,
                "character": 12
            }
        },
        "selectionRange": {
            "start": {
                "line": 17,
                "character": 6
            },
            "end": {
                "line": 17,
                "character": 17
            }
        },
        "detail": null,
        "children": [
            {
                "name": "__init__",
                "kind": 6,
                "range": {
                    "start": {
                        "line": 18,
                        "character": 4
                    },
                    "end": {
                        "line": 19,
                        "character": 12
                    }
                },
                "selectionRange": {
                    "start": {
                        "line": 18,
                        "character": 8
                    },
                    "end": {
                        "line": 18,
                        "character": 16
                    }
                },
                "detail": null,
                "children": [],
                "deprecated": false
            }
        ],
        "deprecated": false
    }
]
```
Notice the `"detail": null`, looks like VS Code does not like that. Outline does not show up in VS Code UI when `"detail": null`.

Removing that `null` fields seems fix the issue with Outline not showing. This is the same case with CodeActions as well where the optional fields get `null` and it does not show up in VS Code UI.

Typical CodeAction response:
```json
```json
{
        "title": "Extract expression into function 'func_njmjvxyh'",
        "kind": "refactor.extract",
        "diagnostics": null,
        "edit": {
            "changes": null,
            "documentChanges": [
                {
                    "textDocument": {
                        "uri": "file:///c:/GIT/repro/lsptest/tests/test_math.py",
                        "version": null
                    },
                    "edits": [
                        {
                            "range": {
                                "start": {
                                    "line": 18,
                                    "character": 8
                                },
                                "end": {
                                    "line": 18,
                                    "character": 8
                                }
                            },
                            "newText": "func_njmjvxyh(self):\n        \r\n        return \n\n    def "
                        },
                        {
                            "range": {
                                "start": {
                                    "line": 19,
                                    "character": 12
                                },
                                "end": {
                                    "line": 19,
                                    "character": 14
                                }
                            },
                            "newText": " = self.func_njmjvxyh()\n"
                        }
                    ]
                }
            ]
        },
        "command": null
    }
```

Related: https://github.com/microsoft/vscode/issues/115793
Related: https://github.com/pappasam/jedi-language-server/issues/60

**Comment by krassowski** [comment]
Already tracked in https://github.com/openlawlibrary/pygls/issues/124 I believe. The consensus appears to be that nulls are not allowed unless explicitly specified in the protocol.

**Comment by danixeee** [comment]
@karthiknadig 

As @krassowski said, we are aware of that and the bug should be fixed in #139. It will be merged and released soon, but there will be backwards-incompatible changes.

**Comment by danixeee** [comment]
Solved in #139.
